### PR TITLE
plugin/pkg: fix staticcheck warning

### PR DESCRIPTION
**Pull request by hwdef**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

Fix static check failures 

**Which issue(s) this PR fixes**:

Ref: #81657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: plugin/pkg/admission/storage/persistentvolume/label/admission.go [hunk 1]
```diff
@@ -335,7 +335,7 @@ func (l *persistentVolumeLabel) getAzurePVLabeler() (cloudprovider.PVLabeler, er
 
 		azurePVLabeler, ok := cloudProvider.(cloudprovider.PVLabeler)
 		if !ok {
-			return nil, errors.New("Azure cloud provider does not implement PV labeling")
+			return nil, errors.New("azure cloud provider does not implement PV labeling")
```
Comment (neolit123) [inline plugin/pkg/admission/storage/persistentvolume/label/admission.go]: can be change to the following to preserve the capitalization of the cloud provider name:
> the Azure cloud provider...

**Comment by neolit123** [comment]
there seem to be related CI failures here:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85633/pull-kubernetes-bazel-test/1199313218258866177

**Comment by neolit123** [comment]
try reproducing the unit test failures locally using:
```
cd kubernetes
go test ./plugin/pkg/admission/storage/...
```

**Comment by hwdef** [comment]
image

the result is the same as CI , I do not know how to fix it, 
Cloud you please help me fix it?
Thank you very much

**Comment by neolit123** [comment]
maybe the unit tests try to match exact string for the errors and now this PR has changed it?

File: plugin/pkg/admission/storage/persistentvolume/resize/admission.go [hunk 2]
```diff
@@ -99,7 +99,7 @@ func (pvcr *persistentVolumeClaimResize) Validate(ctx context.Context, a admissi
 	}
 
 	if oldPvc.Status.Phase != api.ClaimBound {
-		return admission.NewForbidden(a, fmt.Errorf("Only bound persistent volume claims can be expanded"))
+		return admission.NewForbidden(a, fmt.Errorf("only bound persistent volume claims can be expanded"))
```
Comment (hwdef) [inline plugin/pkg/admission/storage/persistentvolume/resize/admission.go]: @neolit123 
the error information is from here,
but I do not understand why this change produced an error.

Comment (neolit123) [reply plugin/pkg/admission/storage/persistentvolume/resize/admission.go]: it's possibly this, like i mentioned earlier:
> maybe the unit tests try to match exact string for the errors and now this PR has changed it?

Comment (hwdef) [reply plugin/pkg/admission/storage/persistentvolume/resize/admission.go]: Yes,you are right.
I find the string and I changed it.

**Review by SataQiu — COMMENTED** [review]
/lgtm

File: plugin/pkg/admission/noderestriction/admission_test.go [hunk 3]
```diff
@@ -24,7 +24,6 @@ import (
 	"testing"
 	"time"
 
-	corev1 "k8s.io/api/core/v1"
```
Comment (deads2k) [inline plugin/pkg/admission/noderestriction/admission_test.go]: prefer corev1

File: plugin/pkg/admission/gc/gc_admission.go [hunk 4]
```diff
@@ -258,7 +258,7 @@ func newBlockingOwnerDeletionRefs(newObj, oldObj runtime.Object) []metav1.OwnerR
 			ret = append(ret, ref)
 			continue
 		}
-		wasNotBlocking := oldRef.BlockOwnerDeletion == nil || *oldRef.BlockOwnerDeletion == false
```
Comment (deads2k) [inline plugin/pkg/admission/gc/gc_admission.go]: prefer the original code block

File: plugin/pkg/admission/priority/admission.go [hunk 5]
```diff
@@ -32,7 +32,6 @@ import (
 	schedulingv1listers "k8s.io/client-go/listers/scheduling/v1"
 	"k8s.io/component-base/featuregate"
 	"k8s.io/kubernetes/pkg/apis/core"
-	api "k8s.io/kubernetes/pkg/apis/core"
```
Comment (deads2k) [inline plugin/pkg/admission/priority/admission.go]: prefer coreapi

File: plugin/pkg/admission/serviceaccount/admission.go [hunk 6]
```diff
@@ -259,11 +259,7 @@ func shouldIgnore(a admission.Attributes) bool {
 		return true
 	}
 	_, ok := obj.(*api.Pod)
-	if !ok {
```
Comment (deads2k) [inline plugin/pkg/admission/serviceaccount/admission.go]: prefer this explicit phrasing.

**Comment by hwdef** [comment]
@deads2k 
I think some of your suggestion is right.
such as coreapi, corev1
But others , I have different opinions. 
I change it ,because I want to  follow the rules of static checking, Not because of my personal preferences.
In  #81657, we do a lot of work about it.
What do you think?

File: plugin/pkg/admission/gc/gc_admission.go (hunk 4 above)
Comment (thockin) [reply plugin/pkg/admission/gc/gc_admission.go]: I agree.  Is there a way to fix staticcheck?

File: plugin/pkg/admission/noderestriction/admission_test.go (hunk 3 above)
Comment (thockin) [reply plugin/pkg/admission/noderestriction/admission_test.go]: Agree - this is pretty consistent across codebase

**Review by thockin — COMMENTED** [review]
You might break this PR into 2 - parts with no contention and parts that we want to argue about across the codebase.

File: plugin/pkg/admission/gc/gc_admission.go (hunk 4 above)
Comment (hwdef) [reply plugin/pkg/admission/gc/gc_admission.go]: yes

**Comment by hwdef** [comment]
I think I better turn it off.
